src/transformers/models/bigscience176b/modeling_bigscience176b.py

```diff
@@ -69,25 +71,48 @@
 _TOKENIZER_FOR_DOC = "BigScience176BTokenizer"
 
 BIGSCIENCE176B_PRETRAINED_MODEL_ARCHIVE_LIST = [
-    "bigscience/BigScience176B",
+    "bigscience/bigscience-small-testing",
+    # "bigscience/BigScience176B",
     # See all BigScience176B models at https://huggingface.co/models?filter=bigscience176b
 ]
 
+# Utility functions below:
+
 def attention_mask_func(attention_scores, attention_mask):
         return attention_scores.masked_fill_(attention_mask, -10000.0)
 
+def transpose_tensor(x):
+    return x.transpose(0, 1).contiguous()
+
+def bias_dropout_add(x, bias, residual, prob, training):
+    # type: (Tensor, Tensor, Tensor, float, bool) -> Tensor
+    out = torch.nn.functional.dropout(x + bias, p=prob, training=training)
+    out = residual + out
+    return out
+
+
+def get_bias_dropout_add(training):
+    def _bias_dropout_add(x, bias, residual, prob):
+        return bias_dropout_add(x, bias, residual, prob, training)
+    return _bias_dropout_add
+
+@torch.jit.script
+def bias_dropout_add_fused_train(x, bias, residual, prob):
+    # type: (Tensor, Tensor, Tensor, float) -> Tensor
+    return bias_dropout_add(x, bias, residual, prob, True)
+
+
+@torch.jit.script
+def bias_dropout_add_fused_inference(x, bias, residual, prob):
+    # type: (Tensor, Tensor, Tensor, float) -> Tensor
+    return bias_dropout_add(x, bias, residual, prob, False)
+
 class BigScience176BAttention(nn.Module):
-    def __init__(self, config, layer_number=None):
+    def __init__(self, config, layer_number=None, dtype=torch.bfloat16):
```

Comment:
Why not reading the `dtype` from the `config`?

Reply:
It is because I assert that the dtype is done once [here](https://github.com/younesbelkada/transformers/blob/e27e554e15da432d0d0aafb60b554d7c9640bfa9/src/transformers/models/bigscience176b/modeling_bigscience176b.py#L689), I did that to avoid asserting as many times as you retrieve the dtype from the config. If you think that the assert is not necessary, then I can read the dtypes directly from the config files on all the modules

Reply:
Actually I think that it makes sense to retrieve it directly from the config. One assert at the very beginning should be sufficient. Fixed it [here](https://github.com/younesbelkada/transformers/commit/fc3056d0f024bd3f5ccf76bc7a7483628f1eefbf)